checkov/common/checks_infra/solvers/connections_solvers/connection_exists_solver.py

```diff
@@ -25,45 +27,82 @@ def __init__(
+                opposite_vertices = None
+                if origin_attributes in self.vertices_under_resource_types:
+                    opposite_vertices = self.vertices_under_connected_resources_types
+                elif origin_attributes in self.vertices_under_connected_resources_types:
+                    opposite_vertices = self.vertices_under_resource_types
+                if not opposite_vertices:
+                    continue
 
-            destination_block_type = destination_attributes.get(CustomAttributes.BLOCK_TYPE)
-            if destination_block_type == BlockType.OUTPUT:
-                try:
-                    output_edges = graph_connector.edges(v, data=True)
-                    _, output_destination, _ = next(iter(output_edges))
-                    output_destination = graph_connector.nodes(data=True)[output_destination]
-                    output_destination_type = output_destination.get(CustomAttributes.RESOURCE_TYPE)
-                    if self.is_associated_edge(
-                        origin_attributes.get(CustomAttributes.RESOURCE_TYPE), output_destination_type
-                    ):
-                        passed.extend([origin_attributes, output_destination])
-                except StopIteration:
+                destination_attributes = graph_connector.nodes(data=True)[v]
+                if destination_attributes in opposite_vertices:
+                    if origin_attributes in self.excluded_vertices or destination_attributes in self.excluded_vertices:
+                        failed.extend([origin_attributes, destination_attributes])
+                    elif origin_attributes in self.unknown_vertices or destination_attributes in self.unknown_vertices:
+                        unknown.extend([origin_attributes, destination_attributes])
+                    else:
+                        passed.extend([origin_attributes, destination_attributes])
+                    destination_attributes["connected_node"] = origin_attributes
                     continue
+
+                destination_block_type = destination_attributes.get(CustomAttributes.BLOCK_TYPE)
+                if destination_block_type == BlockType.OUTPUT:
+                    try:
+                        output_edges = graph_connector.edges(v, data=True)
+                        _, output_destination, _ = next(iter(output_edges))
+                        output_destination = graph_connector.nodes(data=True)[output_destination]
+                        output_destination_type = output_destination.get(CustomAttributes.RESOURCE_TYPE)
+                        if self.is_associated_edge(
+                            origin_attributes.get(CustomAttributes.RESOURCE_TYPE), output_destination_type
+                        ):
+                            passed.extend([origin_attributes, output_destination])
+                    except StopIteration:
+                        continue
```

Comment:
shouldn't there be similar logic to handle output blocks to the igraph implementation as well?

Reply:
as i remember it was not needed, but if we have a test case and it doesn't work with the current implementation, then we should fix it. Basically `igraph` returns back the whole connection path and not one at a time like `networkx` so our implementation is different between them.